### PR TITLE
Feature/15356 invite link

### PR DESCRIPTION
**Pull request by aerych**
Refs #15356 

This PR wires up the UI, analytics, and associated behaviors for the Invite Links feature.

Changes include:
- An invite links section that appears only for P2s
- An option to generate links if there are none. 
- Options to share, configure, or disable existing links.
- Errors should be displayed in a "toast" notice.
- Tracks events that should match [Android](https://github.com/wordpress-mobile/WordPress-Android/pull/14045)

Note: 
- The share cell is missing an share icon. I didn't want to hold up the PR for this minor detail.  To support the icon, the cell is going to need to be rebuilt and I'd rather encapsulate those changes in a separate PR. 
- I've added a spinner to the generate and disable cells that appears when the respective API calls are being made. 
- I've been a little sneaky and co-opted the availableRoles array to ensure that A) invite links are sorted to match and B) the invite links "Role" selection matches the "Role" selection in section 2. This should be okay as the actual roles should not differ.

To test:

### Non-P2
- [ ] View the invite person screen on a non-p2 site.  Confirm that the invite links section is not shown and no invite links tracks events are dispatched.

### P2
- [ ] View a P2 and confirm you see an option to Generate links.
- [ ] Use the generate feature and confirm that the invite links section updates with options to share and manage links. 
- [ ] Confirm you can change the selected invite link via the Role cell. 
- [ ] Confirm you can share the selected invite link via the share cell. Share options will vary depending on simulator / device.
- [ ] Confirm you can disable links via the disable feature. 
- [ ] Confirm the appropriate tracks events are dispatched. 
- [ ] Confirm the UI matches the design except as noted above.
- [ ] Confirm you see spinners during api requests as expected. 
- [ ] Confirm that row taps are ignored while the feature is busy.

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Examples:
Simulator Screen Shot - iPhone 11 Pro - 2021-03-03 at 16 11 49
Simulator Screen Shot - iPhone 11 Pro - 2021-03-03 at 16 11 51
Simulator Screen Shot - iPhone 11 Pro - 2021-03-03 at 16 11 54
Simulator Screen Shot - iPhone 11 Pro - 2021-03-03 at 16 11 57

### CODE REVIEW

**Comment by aerych** [comment]
cc @mattmiklic for a design check :)

File: WordPress/Classes/ViewRelated/People/InvitePersonViewController.swift [hunk 1]
```diff
@@ -410,6 +560,197 @@ private extension InvitePersonViewController {
...
+                generateInviteLinks()
+            } else {
+                shareInviteLink()
+            }
+        case .role:
+            // Handled by segue
+            break
+        case .expires:
+            // No op.
+            break
```
Comment (ScoutHarris) [inline WordPress/Classes/ViewRelated/People/InvitePersonViewController.swift]: Couldn't `default` cover these?

**Review by ScoutHarris — CHANGES_REQUESTED** [review]
Hey Eric.

It all looks good except for one thing. The Message footer is now blue, and is a link to the roles definition. 

message_footer

File: WordPress/Classes/ViewRelated/People/InvitePersonViewController.swift (hunk 1 above)
Comment (aerych) [reply WordPress/Classes/ViewRelated/People/InvitePersonViewController.swift]: Why yes. Yes it could. 👍

**Review by mattmiklic — CHANGES_REQUESTED** [review]
@aerych This looks good for the most part, but the "Generate new link" and "Share invite link" buttons are a shade of blue I don't recognize. Could you double-check that they're using our `primary` blue (blue 50 from Color Studio)?

**Comment by aerych** [comment]
Thanks @ScoutHarris and @mattmiklic!  
I've updated the colors to be sure they are the correct ones. Please confirm :) 
The message section footer should no longer look or behave like a link.
I've cleaned up the switch statement. 

Ready for another peek!

Simulator Screen Shot - iPhone 11 Pro - 2021-03-04 at 12 28 19

**Review by mattmiklic — APPROVED** [review]
Looks good to me now!

**Review by ScoutHarris — APPROVED** [review]
👍 !

:shipit:

**Comment by aerych** [comment]
Thank you all!!!

**Pull request merged**